src/ray/core_worker/core_worker.cc

```diff
@@ -1220,12 +1216,20 @@ void CoreWorker::Exit(
             not_actor_task = actor_id_.IsNil();
           }
           if (not_actor_task) {
-            // If we are a task, then we cannot hold any object references in the
-            // heap. Therefore, any active object references are being held by other
-            // processes. Wait for these processes to release their references
-            // before we shutdown. NOTE(swang): This could still cause this worker
-            // process to stay alive forever if another process holds a reference
-            // forever.
+            // Normal tasks should not hold any object references in the heap after
```

Comment:
> However, there is another edge case that ReleaseAllLocalReferences does currently handle, which is when a normal task (not an actor task) stores an ObjectRef in a global variable before exiting.

Did you find it out due to a failed test? I doubt `ReleaseAllLocalReferences` is intended to be used to fix the global var issue.

Reply:
OK, let's keep it for now then.